numpy/core/tests/test_argparse.py

```diff
@@ -0,0 +1,62 @@
+"""
+Tests for the private NumPy argument parsing functionality.
+They mainly exists to ensure good test coverage without having to try the
+weirder cases on actual numpy functions but test them in one place.
+
+The test function is defined in C to be equivalent to (errors may not always
+match exactly, and could be adjusted):
+
+    def func(arg1, /, arg2, *, arg3):
+        i = integer(arg1)  # reproducing the 'i' parsing in Python.
+        return None
+"""
+
+import pytest
+
+import numpy as np
+from numpy.core._multiarray_tests import argparse_example_function as func
+
+
+def test_invalid_integers():
+    with pytest.raises(TypeError,
+            match="integer argument expected, got float"):
```

Comment:
Is this test running? It should be triggering code coverage of the new functions (the error message comes from`PyArray_PythonPyIntFromInt`) but coverage claims otherwise. I wonder what is going on?

Reply:
Definitely gets run. Confuses me as well, the best idea I have is that the inclusion/compilation both with `_multiarray_tests` and with `_multiarray_umath` confuses the coverage (it no seems to report no coverage at all)?  Which would mean that `mem_overlap` coverage should also be off, and that sounds a bit familiar.